### ISSUE TITLE
Why do I get lightbulbs/refactorings on all quoted strings (among other things)?

### ISSUE THREAD

**Issue by chrisdias** [Microsoft/vscode#33972]
I seem to get *a lot* of lightbulbs.  For examaple, when I highlight the entire line here why am i getting a refactoring?

![image](https://user-images.githubusercontent.com/1487073/30189265-b3c866c2-93e7-11e7-8ffc-d8c98eeae089.png)

If i put the cursor in the `'docker'` string I get a lightbulb:

![image](https://user-images.githubusercontent.com/1487073/30189322-f24e0bc2-93e7-11e7-9494-38a8ce71840a.png)

I move the cursor to `getConfiguration` and the lightbulb goes away...

![image](https://user-images.githubusercontent.com/1487073/30189333-03ed66e8-93e8-11e7-88b2-d94702219d34.png)

Actually, it looks like no matter where I have a quoted string, I get a lightbulb. Here my cursor is in `"Images"`:

![image](https://user-images.githubusercontent.com/1487073/30189358-2bdc7432-93e8-11e7-99b9-70631389ab6f.png)

And when I tell it to refactor my code into a new function, I get this which is technically correct I guess, but not very useful as far as I can tell.

``` TypeScript
    private newFunction(): string {
        return "Images";
    }
```

**Comment by mjbvz** [comment]
This looks like a bug on our side for strings. The lightbulb should only show up when you either have a selection or when the cursor is in a range with a diagnostic of some kind

More broadly, we should really not support extracting literals like this at all. This is tracked upstream by https://github.com/Microsoft/TypeScript/issues/18060

**Comment by mjbvz** [comment]
Looks like insiders only. 1.16 not effected in my testing

@jrieken Was the intent to only keep in the #33241 empty selection fix for 1.16 and then revert it in master? (d8ec15e625cb8f340278e875895d0667a9e28032) If so, I'll push harder to get the TS fix in for TS 2.5.3 and see if we can inline the empty selection logic in the TS extension side for now

**Comment by jrieken** [comment]
> Was the intent to only keep in the #33241 empty selection fix for 1.16 and then revert it in master? (d8ec15e) I

Yes. Having lightbulbs on empty selections is valid, think of a invert-if-else hint at the end of an if-statement,  more here: https://github.com/Microsoft/vscode/issues/33241#issuecomment-325958906

On the good side, the lightbulb oracle now uses the range of the selection (unless it's a diagnostics) and the provider can check how much is selected. E.g. I think it useful to offer 'extract method' when having a single statement selected, it should be at least 2 statements.
